test/less/parse-interpolation.less

```diff
@@ -0,0 +1,29 @@
+@inputs: input[type=text], input[type=email], input[type=password], textarea;
+
+@{inputs} {
+  &:focus {
+    foo: bar;
+  }
+} 
+
+@classes: ~".a, .b, .c";
+
+@{classes} {  
```

Comment:
What about more complex tests like:
```less
.bar {
    foo-@{inputs}&:hover, baz {...}
}
```
etc?

Reply:
I had this working perfectly with: 
```less
.bar {
    foo-@{classes}&:hover, baz {...}
}
```
and more complex cases, and then I tried....
```less
@c: ~'.a, .b';
@d: ~' > .c';
@e: ~' + .d';

@{c}@{d}@{e} {
  foo: bar;
}
```
...aaaaaand wasted the rest of my weekend lol.

The problem was I had a kind of "selector merging" behavior with something like:
```less
@textClasses: ~'[class="text"], .text';

input@{textClasses} {
  background: red;
}

//outputting
input[class="text"],
input.text {
  background: red;
}
```
That is, if we assume that interpolated text should be merged into the selector list, then it stands to reason that it should merged with any adjacent elements and combinators.

All of that I got working, but naïvely. As soon as there were multiple vars containing multiple selectors, elements, and combinators, it borked. Not only would it be inconsistent behavior, it's not clear what the expected behavior SHOULD be.

Maybe a more rational approach is to try to stringify everything in the selector, including the eval'd variable, THEN re-parse the entire selector list, rather than try to parse the variable and merge. That would not allow that `input[class="text"], input.text` output, but that seems to have add-on misery with multiple vars and combinators.

I dunno. My brain has died now.